modules/040-node-manager/images/bashible-apiserver/pkg/template/steps_storage.go

```diff
@@ -138,3 +289,99 @@ func readTemplates(rootDir string, templates map[string][]byte) error {
 		return nil
 	})
 }
+
+func (s *StepsStorage) AddNodeGroupConfiguration(nc *NodeGroupConfiguration) {
+	name := fmt.Sprintf("%d_%s", nc.Spec.Weight, nc.Name)
+
+	bundle := "*"
+	if nc.Spec.Bundle != "" {
+		bundle = nc.Spec.Bundle
+	}
+
+	nodeGroups := []string{"*"}
+	if len(nc.Spec.NodeGroups) > 0 {
+		nodeGroups = nc.Spec.NodeGroups
+	}
+
+	sc := nodeConfigurationScript{
+		Name:    name,
+		Content: nc.Spec.Content,
+	}
+
+	s.m.Lock()
+	defer s.m.Unlock()
+	for _, ng := range nodeGroups {
+		key := fmt.Sprintf("%s:%s", bundle, ng)
```

Comment:
Instead of concatenating strings, what do you think about using struct with two fields (`bundle` and `ng`) for `nodeGroupConfigurations` keys (and the same for system scripts)?

Reply:
Nope. Atm I want to see string keys here :)